src/ncp-spinel/SpinelNCPInstance-DataPump.cpp

```diff
@@ -299,6 +303,16 @@ SpinelNCPInstance::ncp_to_driver_pump()
 			goto on_error;
 		}
 
+#if OPENTHREAD_ENABLE_NCP_SPINEL_ENCRYPTER
+		size_t inboundFrameTransformedLen = sizeof(mInboundFrame);
+		if (!SpinelEncrypter::DecryptInbound(mInboundFrame, mInboundFrameSize, mInboundFrame, &inboundFrameTransformedLen))
```

Comment:
We are passing the same `mInBoundFrame` buffer as both input and output buffer to `DecryptInbound()`  assuming its implementation can handle this correctly. If that's the case, how about if we change the `DecryptInbound` to only accept one buffer pointer and does in-place description. This way it forces the implementer to be careful about this.

I am fine with the `DecryptInbound()` to expect an input and output buffer. In that case, I would suggest we send it two different buffers. How about if we have something like this (use a local buffer to store transformed frame - since on wpantund we are not as resource-limited)....
```
#if OPENTHREAD_ENABLE_NCP_SPINEL_ENCRYPTER
        size_t inboundFrameTransformedLen = sizeof(mInboundFrame);
        uint8_t inboundFrameTransforemed[SPINEL_FRAME_BUFFER_SIZE];
        if (!SpinelEncrypter::DecryptInbound(mInboundFrame, mInboundFrameSize, inboundFrameTrasnformed, &inboundFrameTransformedLen))
        {
        ....
        }
        memcpy(mInboundFrame, inboundFrameTransformed, inboundFrameTrasnformedLen);
        mInboundFrameSize = inboundFrameTrasnformedLen;
```

Reply:
Sounds good. Thanks.

Reply:
Done.